File: nodes/string-replace/string-replace.process.js
export default async ({inputs, settings, config}) => {

  const text = String(inputs.text || "");
  const search = String(inputs.search || "");
  const replacement = String(inputs.replacement || "");
  
  if (settings.regex) {
    const flags = settings.case_sensitive ? 'g' : 'gi';
    const regex = new RegExp(search, flags);
    return { text: text.replace(regex, replacement) };
  }
  
  if (!settings.case_sensitive) {
    const regex = new RegExp(search.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'), 'gi');
    return { text: text.replace(regex, replacement) };
  }
  
  return { text: text.replaceAll(search, replacement) };
}; 

File: nodes/string-replace/string-replace.process.py
import re

async def process(inputs, settings, config, nodeConfig):
  text = str(inputs.get("text", ""))
  search = str(inputs.get("search", ""))
  replacement = str(inputs.get("replacement", ""))
  
  if settings.get("regex", False):
    flags = 0 if settings.get("case_sensitive", True) else re.IGNORECASE
    return {
      "text": re.sub(search, replacement, text, flags=flags)
    }
  
  if not settings.get("case_sensitive", True):
    search = re.escape(search)
    return {
      "text": re.sub(search, replacement, text, flags=re.IGNORECASE)
    }
    
  return {
    "text": text.replace(search, replacement)
  } 